zookeeper-server/src/test/java/org/apache/zookeeper/server/auth/znode/groupacl/X509ZNodeGroupAclProviderTest.java

```diff
@@ -266,6 +266,20 @@ public void testConnectionFiltering() {
     provider
         .handleAuthentication(new ServerAuthenticationProvider.ServerObjs(zks, closableMockServerCnxn), new byte[0]);
     Assert.assertTrue(closableMockServerCnxn.isClosed());
+    System.clearProperty(X509AuthenticationConfig.DEDICATED_DOMAIN);
+
+    // Non-authorized user (connection filtering not enforced)
+    closableMockServerCnxn = new ClosableMockServerCnxn();
+    closableMockServerCnxn.clientChain = new X509Certificate[]{domainXCert};
+    System.setProperty(X509AuthenticationConfig.DEDICATED_DOMAIN, "DomainY");
+    System.setProperty(X509AuthenticationConfig.ENFORCE_DEDICATED_DOMAIN, "false");
```

Comment:
do you want to add any test cases for  Non-authorized user with connection filtering enforced? May be you already have that test case. Sorry, if it is already there and i didn't see